sql/firefox_accounts/fxa_amplitude_email_clicks/view.sql

```diff
@@ -0,0 +1,220 @@
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.firefox_accounts.fxa_amplitude_email_clicks`
+AS
+WITH hmac_key AS (
+  SELECT
+    AEAD.DECRYPT_BYTES(
+      (SELECT keyset FROM `moz-fx-dataops-secrets.airflow_query_keys.fxa_prod`),
+      ciphertext,
+      CAST(key_id AS BYTES)
+    ) AS value
+  FROM
+    `moz-fx-data-shared-prod.firefox_accounts_derived.encrypted_keys_v1`
+  WHERE
+    key_id = 'fxa_hmac_prod'
+),
+sendjobs_numbered AS (
+  SELECT
+    *,
+    ROW_NUMBER() OVER (PARTITION BY SendID ORDER BY snapshot_date DESC) AS _n
+  FROM
+    `mozilla-cdp-prod.sfmc.sendjobs`
+),
+sendjobs AS (
+  SELECT
+    * EXCEPT (_n)
+  FROM
+    sendjobs_numbered
+  WHERE
+    _n = 1
+),
+customers_numbered AS (
+  SELECT
+    *,
+    ROW_NUMBER() OVER (PARTITION BY email ORDER BY fxa_ts DESC) AS _n
+  FROM
+    `mozilla-cdp-prod.fxa.customer_record`
+),
+customers_deduped AS (
+  SELECT
+    * EXCEPT (_n),
+    IF(
+      net.public_suffix(email) IS NULL,
+      NULL,
+      rtrim(
+        substr(net.host(email), 1, length(net.host(email)) - length(net.public_suffix(email))),
+        '.'
+      )
+    ) AS email_provider
+  FROM
+    customers_numbered
+  WHERE
+    _n = 1
+),
+-- We filter out email providers with fewer than 10k customers in this dataset,
+-- roughly similar to how we filter out cities with fewer than 10k residents for
+-- GeoIP lookups in the data pipeline.
+email_providers_over_10k AS (
+  SELECT
+    email_provider
+  FROM
+    customers_deduped
+  GROUP BY
+    1
+  HAVING
+    count(*) > 10000
+),
+customers AS (
+  SELECT
+    * EXCEPT (email_provider),
+    TO_HEX(
+      `moz-fx-data-shared-prod`.udf.hmac_sha256((SELECT * FROM hmac_key), CAST(fxa_id AS BYTES))
+    ) AS user_id,
+    email_providers_over_10k.email_provider,
+  FROM
+    customers_deduped
+  LEFT JOIN
+    email_providers_over_10k
+  USING
+    (email_provider)
+)
+SELECT
+  UNIX_MILLIS(clicks.EventDate) AS `time`,
```

Comment:
The amplitude operator looks for a `submission_timestamp` field, in order to just pull in the most recent events. If that's the `time` field here then we can add add it also as `submission_timestamp`.